### DISCUSSION TITLE
Writing a port but dependency name mismatch

### DISCUSSION THREAD

**Discussion by jdx-john** [microsoft/vcpkg#20061] [Q&A]
I am creating a port for https://github.com/alanxz/SimpleAmqpClient. It has dependencies on boost-chrono and on a library called Rabbitmqc (by the same author) https://github.com/alanxz/rabbitmq-c

The author (or someone) previously contributed a port for rabbitmq-c, _librabbitmq_. However when I add this to my port dependencies:

```
{
  "name": "simple-amqp-client",
  "version": "2.5.1",
  "description": "SimpleAmqpClient is an easy-to-use C++ wrapper around the rabbitmq-c C library",
  "homepage": "https://github.com/alanxz/SimpleAmqpClient",
  "dependencies": [
    "boost-chrono",
	"librabbitmq"
  ]
}

```

I still get an error: **Required library Rabbitmqc NOT FOUND**

I'm a total noob at CMake and writing ports but it seems the issue is surely down to naming differences? In vcpkg it's called `librabbitmqc`, but the CMake script for SimpleApqClient [has](https://github.com/alanxz/SimpleAmqpClient/blob/e474a062da90e92e459f32a69d2680c975d3a3b2/CMakeLists.txt#L56):

```
set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/Modules)
find_package(Rabbitmqc REQUIRED)
INCLUDE_DIRECTORIES(SYSTEM ${Rabbitmqc_INCLUDE_DIRS})
```

I'm sure I could hack this for our local use but I'd like to contribute the port, however I have no idea a proper fix. Is some sort of patch required, or is it possible to remap/rename in my portfile somehow?

edit: There are already issues and PR referencing this on the upstream library:
https://github.com/alanxz/SimpleAmqpClient/issues/290
https://github.com/alanxz/SimpleAmqpClient/pull/291

**Comment by Osyotr** [comment]
SampleApqClient uses Find-module with incorrect name (see https://github.com/alanxz/SimpleAmqpClient/blob/master/Modules/FindRabbitmqc.cmake).
Proper fix would be to create a patch to use correct name (afaik librabbitmq exports cmake targets) in `simple-ampq-client` port and submit it upstream.

**Reply by jdx-john** [reply]
Yes sorry I totally forgot to mention that there are actually outstanding issues on that project to do with this, but they are months old. There might even be a PR with the fix, I think.

Forgive my ignorance but I am not familiar how to create and use a patch in this context for vcpkg. Is there an example?
I assume we mean I make the changes locally and 'patch' means a Git patch file for my changes, it's not some cmake-specific meaning of 'patch'?

**Reply by Osyotr** [reply]
Indeed, there is an [example](https://github.com/microsoft/vcpkg/blob/master/docs/examples/patching.md) in docs. You can see more real word examples by inspecting ports directory, i.e. [ports/kf5archive](https://github.com/microsoft/vcpkg/tree/master/ports/kf5archive) or [ports/zlib](https://github.com/microsoft/vcpkg/tree/master/ports/zlib).
And yes, patch is just a diff.

**Reply by dg0yt** [reply]
I'm not convinced that there is an issue with the `FindRabbitmqc.cmake` file name if it is used via `find_package(Rabbitmqc)`. However this module's implementation is far from a usual Find module's structure. That's a really difficult starting point for "a total noob at CMake".

**Reply by jdx-john** [reply]
@dg0yt this upstream PR appears to focus on the same topic: https://github.com/alanxz/SimpleAmqpClient/issues/290? I guess I can try that. It's frustrating because this library is very small/simple, it's only this single dependency which is a problem. Guess I'll see what I can do :)

**Reply by dg0yt** [reply]
@jdx-john This upstream *issue* suggests to switch from the Find module to the exported config. This is the preferred way of finding packages in Modern CMake. 
There is a separate PR https://github.com/alanxz/SimpleAmqpClient/pull/291 which implements this change (and others). You can try to use that PR's diff as a patch:
https://patch-diff.githubusercontent.com/raw/alanxz/SimpleAmqpClient/pull/291.diff

**Reply by jdx-john** [reply]
@dg0yt yes indeed, my mistake on the issue/PR number. This seems to address the core problem (there are other things to fix) - creating a patch from that PR. I can't mark your suggestion as the answer though unless you care to repost it (not sure anyone cares in a discussion)

**Comment by dg0yt** [comment]
Port names in vcpkg (tight rules) have no literal relation to cmake package names.
